tests/unit/lms/services/assignment_test.py

```diff
@@ -239,55 +239,52 @@ def test_is_member(self, svc, db_session):
         assert svc.is_member(assignment, user.h_userid)
         assert not svc.is_member(assignment, other_user.h_userid)
 
-    def test_get_assignments(self, svc, db_session):
-        assert db_session.scalars(svc.get_assignments()).all()
-
-    def test_get_assignments_by_course(self, svc, db_session, assignment):
+    @pytest.mark.parametrize("instructor_h_userid", [True, False])
+    @pytest.mark.parametrize("course_id", [True, False])
+    @pytest.mark.parametrize("h_userids", [True, False])
+    def test_get_assignments(
```

Comment:
This scales as `2^N` where `N` is the number of independent filters. This is OK as long as `N` is small!